toolchain/check/eval.cpp

```diff
@@ -1355,6 +1355,26 @@ static auto TryEvalInstInContext(EvalContext& eval_context,
     case SemIR::ValueAsRef::Kind:
       break;
 
+    case CARBON_KIND(SemIR::SymbolicBindingPattern bind): {
```

Comment:
Maybe I'm misunderstanding -- I though the conclusion of [this discussion](https://discord.com/channels/655572317891461132/655578254970716160/1289359153243885659) was that we'd have the name of the binding always resolve to the `Bind*Name` instruction, not to the `BindingPattern`.

Reply:
Aha, I see. I had missed [this comment](https://discord.com/channels/655572317891461132/655578254970716160/1285628379202064425).

I was expecting that the end result of the decision was that we would never use `SymbolicBindingPattern` as a value or evaluate it as an expression. I think we should try to reach that point, but I also think it'd be OK to not do so in this PR. However, it looks like we've got a fair few changes here supporting constant evaluation of `SymbolicBindingPattern`s, and it might be easier (and closer to what I think the end goal would be) to special-case `SymbolicBindingPattern`s in deduction more directly.

Right now, deduction bails out when it reaches an instruction that doesn't have a symbolic constant value simply because it's a quick way to detect a case where there's nothing to deduce inside, but that check becomes incorrect once we're deducing against patterns. You could just remove it, or alternatively check whether either there's a symbolic constant value or the parameter is a pattern instruction.

Reply:
(Sorry, I should've linked more directly to the relevant part of the discussion.)

As discussed in the meeting, updating deduction to not require this looks a bit too complicated to do in this PR, so I've added a TODO.